db/commitlog/commitlog.cc

```diff
@@ -1675,6 +1679,71 @@ void db::commitlog::segment_manager::flush_segments(uint64_t size_to_remove) {
     totals.bytes_flush_requested += flushing;
 }
 
+void db::commitlog::segment_manager::check_no_data_older_than_allowed() {
+    if (!cfg.commitlog_data_max_lifetime_in_seconds) {
+        return;
+    }
+
+    auto now = gc_clock::now();
+
+    std::unordered_set<cf_id_type> ids;
+    std::optional<replay_position> high;
+
+    for (auto& s : _segments) {
+        auto rp = replay_position(s->_desc.id, db::position_type(s->size_on_disk()));
+        if (rp <= _flush_position) {
+            // already requested.
+            continue;
+        }
+
+        for (auto [id, low_ts] : s->_cf_min_time) {
+            // Ignore CF:s already released.
+            if (!s->_cf_dirty.count(id)) {
+                continue;
+            }
+
+            auto time_since_first_added = std::chrono::duration_cast<std::chrono::seconds>(now - low_ts).count();
+            if (time_since_first_added >= *cfg.commitlog_data_max_lifetime_in_seconds) {
+                // There is data in this segment that has lived longer than allowed (might be flushed actually
+                // but can still affect compaction/resurrect stuff on replay). Collect all dirty 
+                // id:d (stuff keeping this segment alive), and ask for them to be memtable flushed
```

Comment:
nit: `id:s`?

Reply:
should we flush active memtables or only stale ones?

Reply:
Fixed.